### ISSUE TITLE
Difference in symbol sizes using Mathtext with stixsans

### ISSUE THREAD

**Issue by pfigliozzi** [matplotlib/matplotlib#7080]
Environment: Python 2.7.11, Matplotlib 1.5.1, Windows
Installation: Anaconda distribution, or "conda install ..."

I am having trouble with the mathtext rendering for stix font sans-serif. It appears that anything written in mathtext where 'stixsans' is the default is a few font points smaller than the regular font used outside of mathtext even when using the \mathregular{} command. Here is an example below:

```python
import matplotlib
import matplotlib.pyplot as plt
matplotlib.rcParams['mathtext.fontset'] = 'stixsans'
matplotlib.rcParams['font.family'] = 'sans-serif'
matplotlib.rcParams['font.sans-serif'] = 'stixsans'

plt.plot(1,1)
plt.xlabel(r'x ($\mathregular{\mu \pi m}$m)')

```

For the x-axis this yields the following:

![example](http://i.imgur.com/9ELAZ6u.png)

Here both m's are exactly the same but the symbols are italicized. For stix serif the \mathregular would produce a non-italicized mu and pi. However if I want to use these symbols for units in my graph  then I would want them not italicized. I can attempt to straighten them with \mathrm{}:

```python
plt.xlabel(r'x ($\mathregular{\mathrm{\mu \pi m}}$m)')
```

![example](http://i.imgur.com/Yg3WRa9.png)

Now the mu and pi are straight but everything in the \mathrm{} call has shrunk so it is no longer the same size as the regular text. And if you look closely the m in the \mathrm{} is slightly different from the m in the regular font. If I change the order of the calls of \mathregular and \mathrm:

```python
plt.xlabel(r'x ($\mathrm{\mathregular{\mu \pi m}}$m)')
```

![example](http://i.imgur.com/IMQ59In.png)

Now the mu looks alright but perhaps a tad short. The pi is still somehow slanted. But the m in the \mathrm{} looks exactly the same as the one in the regular text.

Does anyone know what is going on here? All I want is to use some of these symbols generated in mathtext to be the stixsans font and be of a similar height as the non mathtext font. Any help would be greatly appreciated.

**Comment by tacaswell** [comment]
attn @zblz 

Could you try the 2.0 beta?

```
conda install -c conda-forge/label/rc -c conda-forge matplotlib
```

and see if this is fixed?  There was a bunch of work done on the font handling (including some work on rm), but I do not know off the top of my head if this would have been affected.

Also, how have you been?

**Comment by pfigliozzi** [comment]
I tried the 2.0 beta and it actually works perfectly. After installing matplotlib 2.0.0b4 I used the following code:

```python
import matplotlib
import matplotlib.pyplot as plt
matplotlib.rcParams['mathtext.fontset'] = 'stixsans'
matplotlib.rcParams['font.family'] = 'sans-serif'
matplotlib.rcParams['font.sans-serif'] = 'stixsans'
fig = plt.figure()
plt.plot(1,1)
plt.xlabel(r'x ($\mathregular{\mu \pi m}$m)')
```

Which produces this result for the text:
![image](http://i.imgur.com/JRHzviY.png)

So it looks like everything is working fine for the stixsans font for mathtext!

I actually just looked at all the new features in matplotlib 2.0 and I am very excited about all the changes especially some of the changes to default styles/colors. However, I just finished coding up all my figures for my first manuscript so I will cautiously try the beta with my plotting scripts to see if it works.

I am doing well, thanks for asking! I am about to submit my first manuscript which will make up the bulk of my thesis. I should only need to do one more letters type manuscript and I should be able to defend my thesis hopefully in winter quarter. How have you been?
